### PR TITLE
fix(scan): a title-stated remote role survives a city-only location

### PR DESCRIPTION
**Pull request by LayerC0de**
## Problem

Some ATSs report the **hiring office** as the location even when the role is remote, and state the remoteness in the **title** instead. Radancy/TalentBrew tenants return bare `City, State` strings:

```
"Program Manager - Remote"   ->   location "Las Vegas, Nevada"
```

An `allow` list written in country/region terms — the natural way to express "US-remote only", and what `portals.example.yml` encourages — then rejects a genuinely remote US role.

**Measured live** on `careers.unitedhealthgroup.com`: 14 PM-family postings, **0 passed `allow`**, 5 of which said "Remote" outright in the title.

The failure is invisible in the counters. These are tallied under *"filtered by location"*, indistinguishable from correctly-rejected on-site roles — so a user sees a plausible-looking number and never learns they lost real leads.

## Fix

`buildLocationFilter()` takes an optional third `title` argument and, as a last resort, accepts an explicit remote marker in the title.

Placed **after** the `block` tier on purpose: a remote title widens `allow` and can **never** rescue an excluded location. `"Program Manager - Remote"` in Bengaluru stays rejected, including via the URL hint.

## Only a work-arrangement marker counts

```js
export const REMOTE_TITLE_RE = /(?<![a-z])remote(?=$|\s*[^a-z\s]|\s+in\b)/;
```

A bare `/remote/` test would admit domain compounds. **"Remote Sensing Program Manager" is an on-site GIS role** — and Esri, a company in the shipped `portals.example.yml`, posts exactly those. So `remote` must be followed by end-of-string, a non-letter (`)`, `,`, `-`), or `" in …"` as in "Remote in MO" — never by another word, which is what forms `remote sensing` / `remote monitoring`.

## Behavior

| title | location | before | after |
|---|---|---|---|
| `… Project Manager - Remote` | `Costa Mesa, California` | ❌ | ✅ |
| `Program Manager - Remote` | `Las Vegas, Nevada` | ❌ | ✅ |
| `… - Remote in MO` | `St Louis, Missouri` | ❌ | ✅ |
| `Project Manager (Remote)` | `Phoenix, Arizona` | ❌ | ✅ |
| `Program Manager - Remote` | `Bengaluru, …, India` | ❌ | ❌ |
| `Project Manager - Remote` | `London, United Kingdom` | ❌ | ❌ |
| `Program Manager - Remote` | Workday URL hint `…/job/Hyderabad-Telangana-India/…` | ❌ | ❌ |
| **`Remote Sensing Program Manager`** | `Redlands, California` | ❌ | ❌ |
| `Senior Project Manager I` (on-site) | `Eden Prairie, Minnesota` | ❌ | ❌ |
| `Program Manager` | `United States` | ✅ | ✅ |

## Compatibility

The new argument is optional, so any caller that omits it keeps the previous location-only semantics. Threaded through all four call sites — one in `scan.mjs`, three in `scan-ats-full.mjs`.

## Real-world effect

On a live scan of an Optum board wired via `provider: radancy`, new offers went from **0 → 15**, including:

- Release Train Engineer — Commercial Payments, Optum Financial — Remote
- Sr. PBM Client Implementation Project Manager — Remote
- Medicaid Pharmacy Network Program Manager — Remote
- IT Manager, Epic Professional Billing — Remote
- Product Capability Manager — AI Transformation, UMR — Remote

All are remote roles that the location filter had been discarding.

## Tests

4 new cases (24–27) in the existing `Location filter` block: the recovery, block-still-wins including the URL hint, the domain-compound guard, and inertness for absent/non-string/blank titles.

Full suite on this branch: **2,452 passed, 0 failed**.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Location filtering can now use job titles to treat roles as “remote” when location details are insufficient, including for undated/unenriched postings.
* **Bug Fixes**
  * Remote-title detection is conservative: it requires unambiguous “remote” markers, excludes negations like “non-remote”/“not remote,” and doesn’t widen regions blocked by earlier rules.
* **Tests**
  * Added assertions for remote-title positives, negation edge cases (including separator variations), and avoiding false matches for “remote”-like domain/compound words.

### CODE REVIEW

**Comment by LayerC0de** [comment]
Companion PR: #2320 (Radancy legacy markup + JSON fragment transport). That PR is what surfaced this bug: its tenants return bare `City, State` locations, which is how the title-stated-remote false negative became measurable (14 PM-family postings, 0 passing `allow`).

File: scan.mjs [hunk 1]
```diff
@@ -225,15 +226,38 @@ export function locationHintFromUrl(url) {
...
+// on-site GIS role, and Esri (a tracked company) posts exactly those. So
+// "remote" must be followed by end-of-string, a non-letter (")", ",", "-"),
+// or " in …" as in "Remote in MO" — never by another word, which is what makes
+// "remote sensing" / "remote monitoring" compounds.
+export const REMOTE_TITLE_RE = /(?<![a-z])remote(?=$|\s*[^a-z\s]|\s+in\b)/;
+
+/** @param {unknown} title @returns {boolean} whether the title marks the role remote. */
+export function titleSignalsRemote(title) {
+  if (typeof title !== 'string' || title.trim() === '') return false;
+  return REMOTE_TITLE_RE.test(title.toLowerCase());
```
[the remark this answers is not in the document]
Comment (LayerC0de) [reply scan.mjs]: Confirmed and fixed in 7a1f778 — thanks, this was a real hole.

Reproduced before changing anything:

```
titleSignalsRemote('Project Manager - Non-Remote')  ->  true
filter('Eden Prairie, Minnesota', undefined, 'Project Manager - Non-Remote')
                                                    ->  admitted
```

Exactly as described: the delimiter clears the `(?<![a-z])` lookbehind and the
trailing position clears the lookahead, so an explicitly on-site role could
bypass a non-empty `allow` list — the inversion of what this tier exists for.
The original guard only inspected what *follows* "remote" (to reject the
"Remote Sensing" domain compound); a negation sits before it.

Fix disqualifies the title before the marker is consulted:

```js
export const REMOTE_NEGATED_RE = /\b(?:non|not|no)[\s-]*remote/;
```

`[\s-]*` spans only spaces and hyphens, so a word-initial "non"/"not" in an
unrelated token cannot reach across to "remote". That boundary is pinned by
tests, since an over-eager guard would quietly reintroduce the false negatives
this PR set out to fix:

| title | remote? |
|---|---|
| `Project Manager - Non-Remote` | no |
| `Project Manager - Not Remote` | no |
| `Office Manager (Non-Remote)` / `- NonRemote` / `- No Remote` | no |
| `Nonprofit Program Manager - Remote` | yes |
| `Not-for-Profit Program Manager - Remote` | yes |
| `Nordic` / `Notary` … `- Remote` | yes |
| `Remote Sensing Program Manager` | no (unchanged) |

A negation anywhere in the title disqualifies it. Over-rejecting is the safe
direction here: this tier only ever *rescues* a posting, so a false negative
restores the previous behavior while a false positive admits an on-site role.

Tests added: 27a covers the negated forms at both the predicate and filter
level; 27b pins the four over-reach cases. Full suite 2454 passed, 0 failed.

File: scan.mjs [hunk 2]
```diff
@@ -225,15 +226,52 @@ export function locationHintFromUrl(url) {
...
+// safe direction: this tier only ever *rescues* a posting, so a false negative
+// restores the previous behavior while a false positive admits an on-site role.
+export const REMOTE_NEGATED_RE = /\b(?:non|not|no)[\s-]*remote/;
+
+/** @param {unknown} title @returns {boolean} whether the title marks the role remote. */
+export function titleSignalsRemote(title) {
+  if (typeof title !== 'string' || title.trim() === '') return false;
+  const lower = title.toLowerCase();
+  if (REMOTE_NEGATED_RE.test(lower)) return false;
+  return REMOTE_TITLE_RE.test(lower);
```
[the remark this answers is not in the document]
Comment (LayerC0de) [reply scan.mjs]: Confirmed and fixed in f19c479 — you were right, and this was the same hole one
layer down.

`REMOTE_NEGATED_RE` used an ASCII-only `[\s-]*` separator while the marker regex
accepts any non-letter as a delimiter, so every non-ASCII dash walked past the
guard. Measured before the fix:

| separator | `titleSignalsRemote('PM - Non<sep>Remote')` |
|---|---|
| ASCII hyphen `-` | `false` — correct |
| en dash `–` U+2013 | **`true` — leak** |
| em dash `—` U+2014 | **`true` — leak** |
| non-breaking hyphen `‑` U+2011 | **`true` — leak** |
| figure dash `‒` U+2012 | **`true` — leak** |
| minus `−` U+2212 | **`true` — leak** |

So an explicitly on-site role could still bypass a non-empty `allow` list purely
by being typeset with a real dash — which ATS titles routinely are.

Fix widens the separator to `[^a-z]*`, matching the marker's own breadth, so no
punctuation variant can slip between the negation and the word. It cannot
over-reach, because it never crosses a letter: in "Nonprofit Program Manager -
Remote" the run after `non` begins with `profit`, so the negation can't reach
`remote`. Verified still-remote for Nonprofit / Not-for-Profit / Nordic /
Notary, and "Remote Sensing"/"Remote Monitoring" remain excluded.

Test 27c pins all six dash variants plus the empty, space and slash separators,
so a future narrowing of that class fails loudly. Full suite 2455 pass / 0 fail.

**Comment by santifer** [comment]
The live measurement in the comment (14 PM postings, 0 passing allow, 5 saying Remote right in the title) is what turns a heuristic into a documented fix, @LayerC0de. Paired with your Radancy revival this morning, those 8,600 healthcare postings are now both readable AND filterable. Merged. 🚀
